modules/gatk4/markduplicates/main.nf

```diff
@@ -9,10 +9,12 @@ process GATK4_MARKDUPLICATES {
 
     input:
     tuple val(meta), path(bam)
+    path  fasta
+    path  fasta_fai
 
     output:
-    tuple val(meta), path("*.bam")    , emit: bam
-    tuple val(meta), path("*.bai")    , optional:true, emit: bai
+    tuple val(meta), path("${prefix}"), emit: output
+    tuple val(meta), path("*.bai")    , emit: index
```

Comment:
I would just output the index as `crai` instead of `bai` when `cram` is output as well